hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestContainerEndpoint.java

```diff
@@ -1268,16 +1269,74 @@ public void testGetContainerInsightsNonSCMContainers()
     // delete container Id 1 from SCM
     reconContainerManager.deleteContainer(ContainerID.valueOf(1));
     Response containerInsights =
-        containerEndpoint.getContainerMisMatchInsights();
+        containerEndpoint.getContainerMisMatchInsights(10, 0, "SCM");
+    Map<String, Object> response =
+        (Map<String, Object>) containerInsights.getEntity();
+    long prevKey = (long) response.get("prevKey");
+    assertEquals(0, prevKey);
     List<ContainerDiscrepancyInfo> containerDiscrepancyInfoList =
-        (List<ContainerDiscrepancyInfo>) containerInsights.getEntity();
+        (List<ContainerDiscrepancyInfo>) response.get(
+            "missingContainerList");
     ContainerDiscrepancyInfo containerDiscrepancyInfo =
         containerDiscrepancyInfoList.get(0);
     assertEquals(1, containerDiscrepancyInfo.getContainerID());
     assertEquals(1, containerDiscrepancyInfoList.size());
     assertEquals("OM", containerDiscrepancyInfo.getExistsAt());
   }
 
+
+  @Test
+  public void testGetContainerInsightsNonSCMContainersWithPrevKey()
```

Comment:
Pls check the test case based on above prevKey logic comment.

Reply:
Done!